### ISSUE TITLE
Support for recursive data?

### ISSUE THREAD

**Issue by masinc** [yukinarit/pyserde#287]
Python: 3.10.8
pyserde: 0.9.5 

I defined the following class and an error occurred.

```python
from dataclasses import dataclass
from serde import serde, field
from typing import Optional

@dataclass
class A:
    a: Optional['A'] # = field(skip_if_false=True)

# RecursionError
A = serde(A)
```
The `field` value is commented out, but a `RecurtionError` is raised in both case.

Rust `Serde` worked with similar recursive data structures.
```rust
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct A {
    a: Option<Box<A>>,
}
```

**Comment by yukinarit** [comment]
Hi @masinc, This is only the workaround I could come up with  

```python
from dataclasses import dataclass
from serde import serde, to_dict, from_dict
from typing import Optional
from typing import Generic, TypeVar

T = TypeVar('T')

@serde
@dataclass
class A(Generic[T]):
    a: Optional[T]

a = A(A(A(None)))
print(to_dict(a))
print(from_dict(A[A[A[int]]], {'a': {'a': {'a': None}}}))  # int in A[A[A[int]]] is a fake type
```

Output
```
{'a': {'a': {'a': None}}}
A(a=A(a=A(a=None)))
```

**Comment by yukinarit** [comment]
Published pyserde v0.9.6!
